src/Stratis.Bitcoin/Utilities/DateTimeProvider.cs

```diff
@@ -76,7 +84,7 @@ public virtual long GetTime()
         /// <inheritdoc />
         public virtual DateTime GetUtcNow()
         {
-            return DateTime.UtcNow;
+            return DateTime.UtcNow.AddSeconds(this.systemTimeOffset);
```

Comment:
You should probably add this offset to the other methods (at least until your other PR is merged).

Reply:
how about GetTimeOffset?

Reply:
Do you mean that the systemTimeOffset should be returned by GetTimeOffset method?